azure-go-sdk-conceptual/azure-sdk-go-qs-vm.md

```diff
@@ -0,0 +1,305 @@
+
+## Install the Azure SDK for Go 
+
+<!-- Intentionally does not include header -->
+[!INCLUDE [azure-sdk-go-get](includes/azure-sdk-go-get.md)]
+
+## Create a service principal
+
+In order to log in non-interactively with an application, you need a service principal. Service principals are part of Role-Based Authentication (RBAC) which create a unique user identity as well as a set of permissions restrictions. In order to create a new VM and its resources, you need to have a service principal authorized to create new Azure resources. To create a new service principal with the CLI, run:
+
+```azurecli-interactive
+az ad sp create-for-rbac --name az-go-vm-quickstart
+```
+
+__Make sure__ that you record the `appId` and `password` values in the output. These are used in your application to authenticate with Azure.
+
+For more information on creating and managing Service Principals through the Azure CLI 2.0, see [Create an Azure service principal with Azure CLI 2.0](/cli/azure/create-an-azure-service-principal-azure-cli).
+
+## Get the code
+
+You can get the quickstart code, and all of its dependencies, with `go get`:
+
+```bash
+go get -u -d github.com/azure-samples/azure-sdk-for-go-samples/_quickstart/deploy-vm/...
+```
+
+This code will compile out of the box, but doesn't run correctly until you provide it information about your Azure account and the created service principal. In `main.go` there is a variable, `config`, which contains an `authInfo` struct. This struct needs to have its field values replaced in order to authenticate correctly.
+
+* `SubscriptionID`: Your subscription ID, which can be obtained from the CLI command `az account show --query id -o tsv`
+* `TenantID`: Your tenant ID, which can be obtained from the CLI command `az account show --query tenantId -o tsv`
+* `ServicePrincipalID`: The `appId` value recorded when creating the service principal
+* `ServicePrincipalSecret`: The `password` value recorded when creating the service principal
+
+There are also values you will need to edit in the `vm-quickstart-params.json` file:
+
+* `vm_password`: The password for the VM user account. This must follow the following password restrictions: It must be at least 6 characters in length, and contain 3 of the following: A lowercase letter, an uppercase letter, a number, or a symbol.
+
+## Running the code
+
+Run the quickstart with the `go run` command.
+
+```bash
+cd $GOPATH/src/github.com/azure-samples/azure-sdk-for-go-samples/_quickstart/deploy-vm
+go run main.go
+```
+
+If there is a failure in the deployment, you will get a message indicating that there was an issue, but without any specific details. Using the Azure CLI, you get the details of

```

Comment:
Add in quickstart code so that this is avoidable and errors are instead printed as part of the application run. This demonstrates good practice __and__ avoids a confusing user interaction step.